doc: add getting started (#4)
have-fun-was-taken/yafc-ce: README.md

Among other things YAFC has Never Enough Items, which is FNEI on steroids. In addition to showing the recipes, it shows which ones you probably want to use, and how much.

> ## Getting Started

Reviewer: I think if we want to expand on how to get started with YAFC, we want to work on the [Download](https://github.com/have-fun-was-taken/yafc-ce/blob/0aace81ce5b9f1d3d91054b4598db2845a64803c/README.md?plain=1#L58) part of Readme.

Essentially, I agree that the Download section can be moved up and changed into the steps you outlined here.

Author: I tried to take your suggestion into account and I've pushed up.  Let me know what you think!

1. Go to the [releases page](https://github.com/have-fun-was-taken/yafc-ce/releases)

> 2. Download the zip file for your given OSX

Reviewer: "OS", not "OSX", because the sentence is about "given Operating System" if I get it correctly.

3. Extract the zip file to your preferred location on your computer

> 4. If on OSX/Linux, make sure to make `YAFC` executable via `chmod +x ./YAFC`

Reviewer 2: I'm pretty sure there package formats around that can preserve file per permissions

Author: True.  Agreed... I'll be honest, I'm not sure how the releases are being set up for this repo :(

Generally, I just run it locally from the repo.  But when I did download the bundle for OSX, I was not able to execute `./YAFC` in the unzipped bundle without the permissions change.

Reviewer 3: Agreed. I guess we need to fix the build script to keep the file permissions. (I checked, after building  with build.sh the permissions are correct)

Reviewer: > I'm not sure how the releases are being set up for this repo :(

So far, it's manual. I ran [build.bat](https://github.com/have-fun-was-taken/yafc-ce/blob/master/build.bat) and uploaded the results. If there are better ways, please let us know.

Author: I added `build.sh` too (for non Windows).  It does build the releases for Windows/Linux/OSX just fine.  And the permissions for `Build/OSX/YAFC` was just fine so... interesting.

I think the setup for now works?  I'd love to see github actions for the releases, but I've yet to dabble with github actions and .NET

Among other things YAFC has Never Enough Items, which is FNEI on steroids. In addition to showing the recipes, it shows which ones you probably want to use, and how much.

## Getting Started

YAFC is a desktop app. Windows build is the most tested, but OSX and Linux are there too. See [Linux and OSX installation instructions](/Docs/LinuxOsxInstall.md).

1. Navigate to the [Releases](https://github.com/have-fun-was-taken/yafc-ce/releases),
2. Download the zip file for your given OS,
3. Extract the zip file to your preferred location,

> 5. Run either `./YAFC` or `./YAFC.exe` (OS dependent) to start YAFC

Reviewer: ```suggestion
4. Run either `./YAFC` or `./YAFC.exe` (OS-dependent),
```

> 6. Once YAFC is opened, make sure to find your mod folder location.  Refer to the [wiki](https://wiki.factorio.com/Application_directory#Locations) for your given OS

Reviewer: ```suggestion
5. Once YAFC is opened, make sure to locate your mod folder. Refer to the [wiki](https://wiki.factorio.com/Application_directory#Locations) for your given OS.
```

Among other things YAFC has Never Enough Items, which is FNEI on steroids. In addition to showing the recipes, it shows which ones you probably want to use, and how much.

> ## **[Download YAFC and Getting Started](https://github.com/have-fun-was-taken/yafc-ce/releases)**

Reviewer: ```suggestion
## Getting Started
```

YAFC is a desktop app. Windows build is the most tested, but OSX and Linux are there too. See [Linux and OSX installation instructions](/Docs/LinuxOsxInstall.md).

> 1. Go to the [releases page](https://github.com/have-fun-was-taken/yafc-ce/releases)

Reviewer: ```suggestion
1. Navigate to the [Releases](https://github.com/have-fun-was-taken/yafc-ce/releases),
```

2. Download the zip file for your given OS,
3. Extract the zip file to your preferred location,

> 4. If on OSX/Linux, make sure to make `YAFC` executable via `chmod +x ./YAFC`

Reviewer: ```suggestion
```
Linux instructions got updated, so this line can be removed. Essentially, every Linux/OSX user is expected to follow the link above for Linux/OSX installation docs.

Reviewer 3: I think it will be better to move this to the Linux installation instructions, instead of fully removing. I'll see if I can make a PR for this without having the repo forked yet (my reuquest to move the 'base repo' is still pending)

> 2. Download the zip file for your given OS
> 3. Extract the zip file to your preferred location on your computer

Reviewer: ```suggestion
2. Download the zip file for your given OS,
3. Extract the zip file to your preferred location,
```
This list is punctuated by commas at the end for the sentences to not be left hanging.
